lighthouse-core/gather/fetcher.js

```diff
@@ -216,17 +224,18 @@ class Fetcher {
           return;
         }
 
-        // Now in the response stage, but the request failed.
-        if (!(responseStatusCode >= 200 && responseStatusCode < 300)) {
-          reject(new Error(`Invalid response status code: ${responseStatusCode}`));
-          return;
-        }
-
-        const responseBody = await this.session.sendCommand('Fetch.getResponseBody', {requestId});
-        if (responseBody.base64Encoded) {
-          resolve(Buffer.from(responseBody.body, 'base64').toString());
+        if (responseStatusCode >= 200 && responseStatusCode <= 299) {
```

Comment:
this block got a fair bit harder to reader, extract to a `_resolveResponseBody` method and replace with

```js
  resolve({status: responseStatusCode, content: this._resolveResponseBody(statusCode, requestId)});
```

?